openeo-geotrellis/src/main/scala/org/openeo/geotrellis/layers/FileLayerProvider.scala

```diff
@@ -1410,18 +1411,54 @@ class FileLayerProvider private(openSearch: OpenSearchClient, openSearchCollecti
+      }
       /**
        * Several edge cases to cover:
        *  - if feature extent is whole world, it may be invalid in target crs
        *  - if feature is in utm, target extent may be invalid in feature crs
-       *  this is why we take intersection
+       *  this is why we take intersection.
+       *  We convert both extents to a common CRS before taking the intersection.
+       *  If one of the CRSes can cover the whole world (non-UTM), this will be used as common CRS.
+       *  We give priority to use the target CRS as common one, because the intersection will be converted to it anyway
        */
-      val targetExtentInLatLon = targetExtent.reproject(feature.crs.get)
-      val featureExtentInLatLon = feature.rasterExtent.get.reproject(feature.crs.get,LatLng)
+      val featureIsUTM = feature.crs.get.proj4jCrs.getProjection.getName == "utm"
+      val targetIsUTM = targetExtent.crs.proj4jCrs.getProjection.getName == "utm"
+      val commonCrs = if (!targetIsUTM) targetExtent.crs
+      else if (!featureIsUTM) feature.crs.get
+      else targetExtent.crs // Avoid conversion imprecision by intersecting directly in the target CRS
+
+      var featureExtentInCommonCRS = safeReproject(featureProjectedExtent, commonCrs)
+      //      if (commonCrs == CRS.fromName("EPSG:4326") && featureExtentInCommonCRS.width > 180) {
+      //        featureExtentInCommonCRS = Extent( // TODO
+      //          featureExtentInCommonCRS.xmax,
+      //          featureExtentInCommonCRS.ymin,
+      //          featureExtentInCommonCRS.xmin + 360,
+      //          featureExtentInCommonCRS.ymax,
+      //        )
+      //      }
+      val targetExtentInCommonCRS = safeReproject(targetExtent, commonCrs)
+      if (!healthCheckExtent(featureExtentInCommonCRS)) {
+        throw new IllegalArgumentException(s"Feature extent $featureExtentInCommonCRS is invalid in common CRS $commonCrs.")
+      }
 
-      val intersection = featureExtentInLatLon.intersection(targetExtentInLatLon).map(_.buffer(1.0)).getOrElse(featureExtentInLatLon)
-      val tmp = expandToCellSize(intersection.reproject(LatLng, targetExtent.crs), theResolution)
+      val intersection = featureExtentInCommonCRS.extent.intersection(targetExtentInCommonCRS.extent).map(_.buffer(1.0))
+      val intersectionTargetCrs = intersection match {
+        case None =>
+          logger.warn(s"Feature extent $featureExtentInCommonCRS and target extent $targetExtentInCommonCRS do not intersect.")
+          // TODO: Discard the feature?
+          targetExtent.extent
+        case Some(value) => value.reproject(commonCrs, targetExtent.crs)
+      }
+      val tmp = expandToCellSize(intersectionTargetCrs, theResolution)
+      if (!healthCheckExtent(ProjectedExtent(tmp, targetExtent.crs))) {
+        throw new IllegalArgumentException(s"Feature extent $featureExtentInCommonCRS is invalid in target CRS ${targetExtent.crs}.")
```

Comment:
This exception message can be more informative: 
* add process id (load_collection/load_stac)
* add feature id
* use STAC terminology, so 'Item' or 'Asset' instead of 'Feature',

Reply:
Reworked to 
`healthCheckExtentAssert(featureExtentInCommonCRS, s"Item extent (${feature.id}) should be valid in common CRS: ")` and `healthCheckExtentAssert(targetExtent, s"Target extent should be valid: ")`

The later will print something like this when it found an issue:
`Target extent should be valid: Extent not within its CRS limits: ProjectedExtent(Extent(649630.0, 1.212245E7, 684180.0, 1.219141E7),EPSG:32631)`

[New PR](https://github.com/Open-EO/openeo-geotrellis-extensions/pull/399/files#diff-891944eb54f3fdfd4e94fc238e95eb5b3817d73b21bc7efa4d95ce3efa44dae2R1429)

For normal usage, I did not see the asserts get triggered tough. They are there just in case. I'll monitor the logs